src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -12761,6 +12766,79 @@ public class C9{}
             Assert.Equal("(System.Int32, System.Int32).C9", m9Tuple.GetTypeMembersUnordered().Single().ToTestDisplayString());
         }
 
+        [Fact]
+        public void CustomValueTupleWithStrangeThings_01_PE()
```

Comment:
>CustomValueTupleWithStrangeThings_01_PE [](http://example.com/codeflow?start=20&length=39)

It would be good to add a test for a scenario like:
```
namespace System
{
    public record struct ValueTuple<T1, T2>(T1 Item1, T2 Item2)
    {
    }
}
```

Right now I get the following, which feels unexpected:
```
error CS0171: Field '(T1, T2).Item1' must be fully assigned before control is returned to the caller
error CS0171: Field '(T1, T2).Item2' must be fully assigned before control is returned to the caller
```

Reply:
Opened [https://github.com/dotnet/roslyn/issues/56327](https://github.com/dotnet/roslyn/issues/56327) to track the unexpected errors.

Reply:
Thanks